github-bot/longhorn_github_bot/__init__.py

```diff
@@ -60,20 +68,156 @@ def issue_transfer(form):
     except ValueError:
         app.logger.warning('could not parse issue_number %s as int', issue_number)
         return
-    pipeline = form.get('to_pipeline_name')
-    if pipeline != ZENHUB_PIPELINE:
-        app.logger.debug('to_pipeline is {}, ignoring'.format(pipeline))
+
+    to_pipeline = form.get('to_pipeline_name')
+    from_pipeline = form.get('from_pipeline_name')
+
+    if to_pipeline == ZENHUB_PIPELINE_ADD_CHECKLIST:
+        add_checklist(issue_number)
+
+    add_to_sprint_pipelines = {ZENHUB_PIPELINE_TRIAGE, ZENHUB_PIPELINE_ICEBOX, ZENHUB_PIPELINE_BACKLOG}
+    remove_from_sprint_pipelines = {ZENHUB_PIPELINE_TRIAGE, ZENHUB_PIPELINE_ICEBOX}
+    if from_pipeline in add_to_sprint_pipelines and to_pipeline not in add_to_sprint_pipelines:
+        add_to_sprint(issue_number)
+    elif from_pipeline not in remove_from_sprint_pipelines and to_pipeline in remove_from_sprint_pipelines:
+        remove_from_sprint(issue_number)
```

Comment:
Why need to check `from_pipeline`?

Reply:
If it's already in `Triage` or `IceBox`. We don't need to do anything.